### PR TITLE
[Basic] Move Dictionary tuple initializer to Basic

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

**Comment by ddunbar** [comment]
@abertelrud any opinions here?

I like this extension because it makes it easy to avoid explicitly using mutable dictionaries in places where you just need to cons up a dictionary from existing data.

File: Sources/Basic/DictionaryExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+extension Dictionary {
+    /// Convenience initializer to create dictionary from tuples.
+    public init<S: Sequence>(items: S) where S.Iterator.Element == (Key, Value) {
+        var result = Dictionary.init()
```
Comment (ddunbar) [inline Sources/Basic/DictionaryExtensions.swift]: Dunno why I wrote it this way, but this should just be `Dictionary()`

Comment (aciidb0mb3r) [inline Sources/Basic/DictionaryExtensions.swift]: Yeah, I noticed that too xD

**Comment by abertelrud** [comment]
Yes, this is a great extension.  Looks good!

**Pull request merged**